test_plans/memcpy2d.asciidoc

```diff
@@ -0,0 +1,151 @@
+:sectnums:
+:xrefstyle: short
+
+= Test plan for 2D memcpy and associated routines that can copy a specified rectangular region in the presence of array padding.
+
+This is a test plan for the APIs described in
+https://github.com/intel/llvm/blob/sycl/sycl/doc/extensions/supported/sycl_ext_oneapi_memcpy2d.asciidoc[SYCL_EXT_ONEAPI_memcpy2d.asciidoc]
+
+
+== Testing scope
+
+=== Device coverage
+
+All of the tests described below are performed only on the default device that
+is selected on the CTS command line.
+
+=== Feature test macro
+
+All of the tests should use `#ifdef SYCL_EXT_ONEAPI_MEMCPY2D` so they can be skipped
+if feature is not supported.
+
+=== Type coverage
```

Comment:
I think it would be OK to use the types listed here only for the functions which are templated on `T`.  For the other functions, it should be sufficient to test only with `unsigned char`.

Reply:
Added type constraint to the _ext_oneapi_memcpy2d_ and _ext_oneapi_memset2d_ functions.